src/main/java/com/elixr/poc/service/UserService.java

```diff
@@ -55,19 +56,23 @@ public ErrorResponse createValidUser(User user) {
      * @param user
      * @return
      */
-    private void saveDataToDatabase(User user) {
+    private User saveDataToDatabase(User user) {
 
         if (user.getId() == null || user.getId().toString().isEmpty()) {
             user.setId(UUID.randomUUID());
         }
-        this.userRepository.save(user);
+        user = this.userRepository.save(user);
+        return user;
     }
 
     /**
-     * Retriving all the users
+     * Finding User by userId and returning the user.
+     * @param userId
      * @return
+     * @throws GlobalException
      */
-    public GetAllResponse getAllUsers() {
-        return GetAllResponse.builder().success(true).users(userRepository.findAll()).build();
+    public User getUserByUserId(UUID userId) throws GlobalException {
+        Optional<User> user = userRepository.findById(userId);
+        return user.orElseThrow(() -> new GlobalException(ApplicationConstants.ID_MISMATCH));
```

Comment:
The error message should match the use case.

Reply:
Updated